realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -628,7 +628,7 @@ private void completedAsyncRealmObject(QueryUpdateTask.Result result) {
                 } else if (compare > 0) {
                     // the caller has advanced we need to
                     // retry against the current version of the caller if it's still empty
-                    if (RealmObject.isValid(proxy)) { // already completed & has a valid pointer no need to re-run
+                    if (RealmObject.isLoaded(proxy)) { // already completed & has a valid pointer no need to re-run
```

Comment:
`isLoaded` guarantees the query completed, it doesn't guarantee a valid pointer (for this we need to call `isValid`)

the instance could be null if the query didn't find anything